### PR TITLE
#159951962 Implement validations for signup endpoint

### PR DESCRIPTION
**Pull request by cmplx-xyttmt**
#### What does this PR do?
Have validations for sign up endpoint working

#### Description of Task to be completed?
Have the signup endpoint `http://127.0.0.1:8000/api/users/` return descriptive error messages when a user provides invalid data.

#### How should this be manually tested?
Go to postman and try supplying the  following invalid data to the `https://ah-backend-tesseract-sta-pr-11.herokuapp.com/api/users/` endpoint:
```
- Leave out some fields in the request
- Input an invalid username. (A username is one which has  more than 4 characters and only has letters, numbers underscores and hyphens)
- Input an invalid password (A valid password has both numbers and characters)
- Input an invalid email
```

#### What are the relevant pivotal tracker stories?
#159951962

### CODE REVIEW

File: authors/apps/authentication/tests/test_validation.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+class ValidationTests(TestCase, BaseTest):
+
+    def setUp(self):
+        self.client = APIClient()
+        self.user = {"user": {"username": "llll",
+                              "email": "l@gmail.com",
+                              "password": "p1234567"
+                              }
+                     }
```
Comment (arthuroe) [inline authors/apps/authentication/tests/test_validation.py]: @cmplx-xyttmt let's define these in the `BaseTest` or use the ones already created there.

Comment (cmplx-xyttmt) [reply authors/apps/authentication/tests/test_validation.py]: @arthuroe , thanks for pointing that out. I've done that.

**Comment by ruganda** [comment]
LGTM

**Comment by byarustev** [comment]
LGTM

**Comment by marthamareal** [comment]
LGTM

**Pull request merged**